src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.Variables.cs

```diff
@@ -0,0 +1,442 @@
+
+            internal static Variables Create(Symbol? symbol)
+            {
+                return new Variables(id: 0, container: null, symbol);
+            }
+
+            internal static Variables Create(VariablesSnapshot snapshot)
+            {
+                var container = snapshot.Container is null ? null : Create(snapshot.Container);
+                var variables = new Variables(snapshot.Id, container, snapshot.Symbol);
+                variables.Populate(snapshot);
+                return variables;
+            }
+
+            private int GetNextId()
+            {
+                return Id +
+#if DEBUG
+                    _nextIdOffset.Next(maxValue: 7) +
+#endif
+                    1;
+            }
+
+            private void Populate(VariablesSnapshot snapshot)
+            {
+                Debug.Assert(_variableSlot.Count == 0);
+                Debug.Assert(_variableTypes.Count == 0);
+                Debug.Assert(_variableBySlot.Count == 1);
+
+                _variableBySlot.AddMany(default, snapshot.VariableSlot.Length);
+                foreach (var pair in snapshot.VariableSlot)
+                {
+                    var identifier = pair.Key;
+                    var index = pair.Value;
+                    _variableSlot.Add(identifier, index);
+                    _variableBySlot[index] = identifier;
+                }
+
+                foreach (var pair in snapshot.VariableTypes)
+                {
+                    _variableTypes.Add(pair.Key, pair.Value);
+                }
+            }
+
+            private Variables(int id, Variables? container, Symbol? symbol)
+            {
+                Debug.Assert(id >= 0);
+                Debug.Assert(id <= IdMask);
+                Debug.Assert(container is null || container.Id < id);
+#if DEBUG
+                _nextIdOffset = container?._nextIdOffset ?? new Random();
+#endif
+                Id = id;
+                Container = container;
+                Symbol = symbol;
+            }
+
+            internal void Free()
+            {
+                Container?.Free();
+                _variableBySlot.Free();
+                _variableTypes.Free();
+                _variableSlot.Free();
+            }
+
+            internal VariablesSnapshot CreateSnapshot()
+            {
+                return new VariablesSnapshot(
+                    Id,
+                    Container?.CreateSnapshot(),
+                    Symbol,
+                    ImmutableArray.CreateRange(_variableSlot),
+                    ImmutableDictionary.CreateRange(_variableTypes));
+            }
+
+            internal Variables CreateNestedMethodScope(MethodSymbol method)
+            {
+                Debug.Assert(GetVariablesForMethodScope(method) is null);
+                Debug.Assert(!(method.ContainingSymbol is MethodSymbol containingMethod) ||
```

Comment:
>!(method.ContainingSymbol is MethodSymbol containingMethod) [](start = 29, length = 59)

Consider simplifying with `methodSymbol.ContainingSymbol is not MethodSymbol containingMethod`

Reply:
See [https://github.com/dotnet/roslyn/issues/49262](https://github.com/dotnet/roslyn/issues/49262).

---
In reply to: [562837613](https://github.com/dotnet/roslyn/pull/50417#discussion_r562837613) [](ancestors = 562837613)